server/src/main/java/org/elasticsearch/repositories/RepositoriesService.java

```diff
@@ -75,6 +86,9 @@
 
     private final Map<String, Repository> internalRepositories = ConcurrentCollections.newConcurrentMap();
     private volatile Map<String, Repository> repositories = Collections.emptyMap();
+    private final RepositoriesStatsArchive repositoriesStatsArchive;
+
+    private volatile long lastKnownClusterVersion;
```

Comment:
I'm not really sure if this is a good idea. This is used to populate the `clusterVersion` field for the active repositories. An alternative would be not populating that field for the active repositories. I'm not really sure.

Reply:
This seems weird indeed. Technically I guess there is a relationship between the CS version and the stats, this is pretty unintuitive. Can't we simply increment a counter whenever we archive and clear any stats instead and use that as our logical time stamp for example?